src/modules/flow/process/output.c

```diff
@@ -152,7 +152,8 @@ watch_start(struct output_data *output)
     flags = fcntl(output->fd, F_GETFL);
     SOL_INT_CHECK(flags, < 0, -errno);
     flags |= O_NONBLOCK;
-    fcntl(output->fd, F_SETFL, flags);
+    if (fcntl(output->fd, F_SETFL, flags) < 0)
+        return -errno;
```

Comment:
I'd rather have a function to set a flag on a file descriptor than replicate this code all over. Maybe `int sol_util_posix_fd_set_flag(int fd, int flag)`?